packages/hardhat-core/src/internal/core/vars/vars-manager.ts

```diff
@@ -0,0 +1,145 @@
+import fs from "fs-extra";
+import debug from "debug";
+import { HardhatError } from "../errors";
+import { ERRORS } from "../errors-list";
+
+interface Var {
+  value: string;
+}
+
+interface VarsFile {
+  _format: string; // Version of the json vars file
+  vars: Record<string, Var>;
+}
+
+const log = debug("hardhat:core:vars:varsManager");
+
+export class VarsManager {
+  private readonly _VERSION = "hh-vars-1";
+  private readonly _ENV_VAR_PREFIX = "HARDHAT_VAR_";
+  private readonly _cache: VarsFile;
+  private readonly _cacheEnv: VarsFile;
+
+  constructor(private readonly _varsFilePath: string) {
+    log("Creating a new instance of VarsManager");
+
+    this._initializeVarsFile();
+    this._cache = fs.readJSONSync(this._varsFilePath);
+
+    this._cacheEnv = this._getVarsFileStructure();
+    this._loadVarsFromEnv();
+  }
+
+  public getStoragePath(): string {
+    return this._varsFilePath;
+  }
+
+  public set(key: string, value: string) {
+    this.validateKey(key);
+
+    const vars = this._readVars();
+
+    vars[key] = { value };
+    this._writeVars(vars);
+  }
+
+  public has(key: string): boolean {
+    return key in this._readVars();
+  }
+
+  public hasWithEnvVars(key: string): boolean {
```

Comment:
why not using a second optional parameter in `has` called `includeEnvVars` or something like that?